nft_ingester/src/batch_mint/batch_mint_persister.rs

```diff
@@ -58,8 +59,11 @@ impl<D: BatchMintDownloader> BatchMintPersister<D> {
         Self { rocks_client, downloader, metrics }
     }
 
-    pub async fn persist_batch_mints(&self, mut rx: Receiver<()>) -> Result<(), JoinError> {
-        while rx.is_empty() {
+    pub async fn persist_batch_mints(
+        &self,
+        cancellation_token: CancellationToken,
+    ) -> Result<(), JoinError> {
+        while !cancellation_token.is_cancelled() {
```

Comment:
`get_batch_mint_to_verify` will block the shutdown

Reply:
yes - it's graceful and we will complete existing work and won't start new iterations or whatever